File: Olympiad/COCI/official/2007/contest4_solutions/skener.cpp

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 4 - Task SKENER
  Programming language C++
*/

#include <cstdio>

#define maxr 50

int r, s, zoomr, zooms;
char grid[maxr][maxr+1];

int main(void) {
   scanf( "%d%d%d%d", &r, &s, &zoomr, &zooms );
   for( int i = 0; i < r; i++ ) scanf( "%s", grid[i] );
   for( int i = 0; i < r*zoomr; i++ ) {
      for( int j = 0; j < s*zooms; j++ ) printf( "%c", grid[i/zoomr][j/zooms] );
      printf( "\n" );
   }
   return 0;
}

File: Olympiad/COCI/official/2007/contest4_solutions/skener.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 4 - Task SKENER
  Programming language Pascal
*)

program skener;

var
    i, j, r, s, zoomr, zooms: integer;
    grid: array[1..50] of string;

begin
    readln( r, s, zoomr, zooms );
    for i := 1 to r do readln( grid[i] );
    for i := 1 to r*zoomr do begin
        for j := 1 to s*zooms do write( grid[ (i-1) div zoomr + 1 ][ (j-1) div zooms + 1 ] );
        writeln;
    end;
end.
